### DISCUSSION TITLE
Problem with integrated with vitepress

### DISCUSSION THREAD

**Discussion by maicss** [giscus/giscus-component#1870] [Q&A] [answered]
Integrated with vitepress, use web-component way, add giscus into default layout as [code here](https://github.com/giscus/giscus-component/blob/main/vue/src/lib/Giscus.vue). But if I don't add `v-pre` to the `giscus-widget` tag, console will give a warning: `[Vue warn]: Failed to resolve component: giscus-widget
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. `. But your component doesn't have this warning, WHY?

**Comment by laymonage** [comment, marked as answer]
https://github.com/giscus/giscus-component/blob/f43d6c27c57efb97835b000c2bdbd32ff56c4364/vue/vite.config.ts#L12

**Reply by maicss** [reply]
Thank you very much. So, it's better use `v-pre` when there is only one webComponnet in my project, right?

**Reply by laymonage** [reply]
I don't know, their web components guide only mention `isCustomElement` and not `v-pre`: https://vuejs.org/guide/extras/web-components.html

**Reply by maicss** [reply]
OK, thanks again, you help me a lot.
